torch/utils/data/dataloader.py

```diff
@@ -946,6 +946,18 @@ def __init__(self, loader):
         else:
             self._data_queue = self._worker_result_queue
 
+        # In some rare cases, persistent workers (daemonic processes)
+        # would be terminated before `__del__` of iterator is invoked
+        # when main process exits
+        # It would cause failure when pin_memory_thread tries to read
+        # corrupted data from worker_result_queue
+        # atexit is used to prevent persistent workers exiting before
+        # pin_memory_thread
+        if self._persistent_workers and self._pin_memory:
```

Comment:
In order to potentially fix part of #71187 for the case of persistent workers, we may want to remove this condition on `self._pin_memory`